src/services/company.js

```diff
@@ -1,19 +1,17 @@
-const Offer = require("../models/Offer");
-
 class CompanyService {
     // TODO
     constructor() {
 
     }
 
-    getOffersInTimePeriod(owner, publishDate, publishEndDate, offer) {
-        const test = offer || Offer;
-        return test.find({
+    getOffersInTimePeriod(owner, publishDate, publishEndDate, OfferModel) {
+        return OfferModel.find({
             owner,
-
-            $or: [{ $and: [{ publishEndDate: { $gte: publishDate } }, { publishEndDate: { $lte: publishEndDate } }] },
-                { $and: [{ publishDate: { $gte: publishDate } }, { publishDate: { $lte: publishEndDate } }] },
-                { $and: [{ publishDate: { $lte: publishDate } }, { publishEndDate: { $gte: publishEndDate } }] }]
+            $or:
+                [{ $and: [{ publishEndDate: { $gte: publishDate } }, { publishEndDate: { $lte: publishEndDate } }] },
```

Comment:
newline between the first `{` and `[` so that all inner "and" conditions are aligned